consumer/processing.go

```diff
@@ -363,27 +413,98 @@ func organizationAllowed(consumer *KafkaConsumer, orgID types.OrgID) bool {
 	return orgAllowed
 }
 
+func verifySystemAttributeIsEmpty(r Report) bool {
+	if r[reportAttributeSystem] != nil {
+		var s system
+		if err := json.Unmarshal(*r[reportAttributeSystem], &s); err != nil {
+			return false
+		}
+		if s.Hostname != "" {
+			return false
+		}
+	}
+	return true
+}
+
+func verifyJSONArrayAttributeIsEmpty(attr string, r Report) bool {
+	if val, exist := r[attr]; exist && val != nil {
+		var arr []interface{}
+		if err := json.Unmarshal(*val, &arr); err != nil {
+			return false
+		}
+		if len(arr) != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// isReportWithEmptyAttributes checks if the report is empty, or if the attributes
+// expected in the report, minus the analysis_metadata, are empty.
+// If this function returns true, this report will not be processed further as it is
+// PROBABLY the result of an archive that was not processed by insights-core.
+// see https://github.com/RedHatInsights/insights-results-aggregator/issues/1834
+func isReportWithEmptyAttributes(r Report, keysToCheck []string) bool {
+	for _, key := range keysToCheck {
+		switch key {
+		case reportAttributeSystem:
```

Comment:
hmm it really feel that some form of Checker interface will be better.

Reply:
Agreed. I'll add that on my todo list =)

Reply:
checker interface added to the PR -> https://github.com/RedHatInsights/insights-results-aggregator/pull/1835/commits/a0f96c59656355e445c48a2e667958abcb60950a